pandas/tests/extension/decimal/array.py

```diff
@@ -131,6 +131,11 @@ def copy(self):
         return type(self)(self._data.copy())
 
     def astype(self, dtype, copy=True):
+        if dtype == self._dtype:
+            if not copy:
+                return self
+            elif copy:
+                return self.copy()
```

Comment:
@jorisvandenbossche Is this implemented correctly? I'm concerned because of the way returning a copy is implemented on line 141, which seems to take into account `dtype.context`. Maybe the fix would be to remove lines 137-138 and let that case fall to 141-142?

Reply:
I think it is correct, yes. The first `is_dtype_equal(dtype, self._dtype)` will only catch decimal dtypes with an equal `context`, and then the `if isinstance(dtype, type(self.dtype)):` will work correctly in case the context differs.

Reply:
Ah, but we might need to pass through the `copy`

(anyway, this array implementation is only for testing, it's not actually used, so it's not super important)

Reply:
Easy enough to check, so I changed it to pass through the `copy`. It passes the test suite the same way, and seems right, so I'll go with that version.